pkg/ottl/functions.go

```diff
@@ -48,15 +50,40 @@ func (p *Parser[K]) newFunctionCall(ed editor) (Expr[K], error) {
 }
 
 func (p *Parser[K]) buildArgs(ed editor, argsVal reflect.Value) error {
-	if len(ed.Arguments) != argsVal.NumField() {
-		return fmt.Errorf("incorrect number of arguments. Expected: %d Received: %d", argsVal.NumField(), len(ed.Arguments))
-	}
-
+	requiredArgs := 0
 	argsType := argsVal.Type()
 
+	unnamedArgs := []argument{}
+	namedArgs := []argument{}
+	i := 0
+
+	for ; i < len(ed.NewArguments); i++ {
+		if ed.NewArguments[i].Name == "" {
+			unnamedArgs = append(unnamedArgs, ed.NewArguments[i])
+		} else {
+			break
+		}
+	}
+	for ; i < len(ed.NewArguments); i++ {
+		if ed.NewArguments[i].Name != "" {
+			namedArgs = append(namedArgs, ed.NewArguments[i])
+		} else {
+			return errors.New("Unnamed argument used after named argument")
+		}
+	}
```

Comment:
We could handled this setup and error in a single loop if we use a bool to keep track of when a named argument has been seen.

Reply:
Done.